src/algorithms/klement.jl

```diff
@@ -0,0 +1,146 @@
+"""
+    Klement(; max_resets = 100, linsolve = NoLineSearch(), linesearch = nothing,
+        precs = DEFAULT_PRECS, alpha = nothing, init_jacobian::Val = Val(:identity),
+        autodiff = nothing)
+
+An implementation of `Klement` [klement2014using](@citep) with line search, preconditioning
+and customizable linear solves. It is recommended to use [`Broyden`](@ref) for most problems
+over this.
```

Comment:
Is that still true? Since the fast polyalg defaults here.

Reply:
Klement is a very interested solver. If it works it converges faster than Broyden but it works in very limited situations. We can consider removing it from the polyalgorithm.